### PR TITLE
fh-sync default handler support

### PR DESCRIPTION
**Pull request by wtrocki**
Implementation for #8

- [x] Fixed unit tests
- [x] Cleanup in dist and various leftovers from fh-js-sdk repo
- [x] Added/changed documentation
- [x] Added default handler for opensource projects

### CODE REVIEW

File: src/cloudHandler.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+                } catch (e) {
+                    return failure(e);
+                }
+                success(responseJson);
+            } else {
+                failure(xhr.responseText);
+            }
+        }
+    };
+    xhr.send(json);
```
Comment (wei-lee) [inline src/cloudHandler.js:36]: this function doesn't send the some of the app info into in the request, like app ids, client device ids etc. see https://github.com/feedhenry/fh-js-sdk/blob/master/src/modules/fhparams.js#L9. Is that deliberate?

Those information is specific to FH cloud apps, so a bit more discussion required here to decide if the default handler need to include the default request handler. Personally I think it shouldn't, and the fh-sync-js client then need to provide APIs to allow override. 

Also if this is the case, then the `cloudPath` should be specified by the user as well. The default `/mbaas/sync/` is also FH only.

Comment (wtrocki) [reply src/cloudHandler.js:36]: @wei-lee 

Thank you for checking that! Sorry for lack of context. This function is not sending this details and that was design decision - this details are for platform elements that aren't part of sync. Default handler will be used only for plain sync users - no platform involved.

>fh-sync-js client then need to provide APIs to allow override.

Yes. setCloudHandler method is part now of api to override handler and use it in fh-js-client. For platform integration we will still use fh.cloud and all related things so nothing changing in that manner. Going to create PR for that soon.  

Default Path can be changed in cloud and client now. "mbaas/sync" will be used when nothing is specified, only to keep it aligned with our example applications. 
I'm open for suggestions to use different path.

TL;DR - from platform point of view all features will be reflected 1:1.
Community (default) handler will be simple and clean to avoid some legacy stuff.

**Review by austincunningham — COMMENTED** [review]
CORS issue , would need to be fixed on the express server side . No changes needed here.

XMLHttpRequest cannot load http://localhost:3000/sync/myShoppingList. Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://localhost:63342' is therefore not allowed access.

**Comment by wtrocki** [comment]
@austincunningham That's just example app issue. We may need to have them together as sample sync js example. Then this problem will not exist. Let's update details in #10

**Comment by wtrocki** [comment]
@austincunningham Error was fixed in fh-sync example application.
See PR: https://github.com/feedhenry/fh-sync/pull/11

File: example/index.html [hunk 2]
```diff
@@ -38,6 +24,7 @@
   //provide listeners for notifications.
   $fh.sync.notify(function(notification){
     var code = notification.code
+    console.log("Sync ", notification);
```
Comment (witmicko) [inline example/index.html:27]: won't this make it too noisy by default?

**Review by witmicko — APPROVED** [review]
one comment, lgtm otherwise.

File: example/index.html (hunk 2 above)
Comment (wtrocki) [reply example/index.html:27]: That's example application. Code is from documentation. We going to move this out of this repo and improve in different ticket.

**Review by austincunningham — APPROVED** [review]
